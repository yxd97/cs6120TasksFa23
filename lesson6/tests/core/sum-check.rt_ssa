@main(n: int) {
.ENTRY:
.ENTRY:
.main_bb0:
  first.1: int = call @sum_by_loop n;
  second.1: int = call @sum_by_formula n;
  isSame.1: bool = eq first.1 second.1;
  print first.1;
  print second.1;
  print isSame.1;
}

@sum_by_loop(n: int): int {
.ENTRY:
.ENTRY:
.sum_by_loop_bb0:
  one.1: int = const 1;
  sum.1: int = const 0;
  i.1: int = const 1;
  sum.2 = id sum.1;
  i.2 = id i.1;
.for_start:
  con.1: bool = le i.2 n;
  br con.1 .for .end;
.for:
  sum.3: int = add sum.2 i.2;
  i.3: int = add i.2 one.1;
  sum.2 = id sum.3;
  i.2 = id i.3;
  jmp .for_start;
.end:
  ret sum.2;
}

@sum_by_formula(n: int): int {
.ENTRY:
.ENTRY:
.sum_by_formula_bb0:
  one.1: int = const 1;
  two.1: int = const 2;
  n_1.1: int = add one.1 n;
  multi.1: int = mul n_1.1 n;
  sum.1: int = div multi.1 two.1;
  ret sum.1;
}

