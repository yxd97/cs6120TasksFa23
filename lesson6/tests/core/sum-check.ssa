@main(n: int) {
.main_bb0:
  n.1: int = id n;
  first.1: int = call @sum_by_loop n.1;
  second.1: int = call @sum_by_formula n.1;
  isSame.1: bool = eq first.1 second.1;
  print first.1;
  print second.1;
  print isSame.1;
}

@sum_by_loop(n: int): int {
.sum_by_loop_bb0:
  n.1: int = id n;
  one.1: int = const 1;
  sum.1: int = const 0;
  i.1: int = const 1;
.for_start:
  i.2 = phi i.1 .sum_by_loop_bb0 i.3 .for;
  sum.2 = phi sum.1 .sum_by_loop_bb0 sum.3 .for;
  con.1: bool = le i.2 n.1;
  br con.1 .for .end;
.for:
  sum.3: int = add sum.2 i.2;
  i.3: int = add i.2 one.1;
  jmp .for_start;
.end:
  ret sum.2;
}

@sum_by_formula(n: int): int {
.sum_by_formula_bb0:
  n.1: int = id n;
  one.1: int = const 1;
  two.1: int = const 2;
  n_1.1: int = add one.1 n.1;
  multi.1: int = mul n_1.1 n.1;
  sum.1: int = div multi.1 two.1;
  ret sum.1;
}

