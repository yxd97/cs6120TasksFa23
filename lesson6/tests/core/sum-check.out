500500
500500
true
